packages/workspace/src/api/methods/registerService.ts

```diff
@@ -1,4 +1,8 @@
+import { ServiceDefinition } from '@scalecube/scalecube-microservice/lib/api';
+
 export interface RegisterServiceRequest {
   serviceName: string;
-  reference: any;
+  definition: ServiceDefinition;
```

Comment:
Why did you removed the reference from this interface ? How do you suggest to pass it if it's not in this request ?

I see that in your external modules you have both `definition` and `reference` :
```
	    const registerServiceData = {
      serviceName: SERVICE_CONFIG.serviceName,
      definition: SERVICE_CONFIG.definition,
    };

     await WORKSPACE.registerService(registerServiceData);
    resolve({ ...registerServiceData, reference: serviceA });
```

Reply:
We need reference only in order to create scalecube Microservice.
I create it in the body of createWorkspace function in WorkspaceFactory. I receive the reference from the bootstrap function, that each service exports (I include reference in the resolve of bootstrap)

Reply:
On my opinion it's not correct. The reference should be part of the register.
We had 2 options, one was to let the workspace register the service, the second was to let the service doing it. We decided to chose the second one, so all the responsibility should be on service side, workspace should just receive result of the promise to know that everything went well